### PR TITLE
perf(pin_api): fetch chunks in parallel

### LINKED ISSUE #4427: POST /pins/ref should fetch and store chunks in parallel

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
closes #4427

### CODE REVIEW

File: pkg/api/pin.go [hunk 1]
```diff
@@ -49,28 +52,60 @@ func (s *Service) pinRootHash(w http.ResponseWriter, r *http.Request) {
 	getter := s.storer.Download(true)
 	traverser := traversal.New(getter)
 
+	sem := semaphore.NewWeighted(100)
+	var errTraverse error
+	var mtxErr sync.Mutex
+	var wg sync.WaitGroup
```
Comment (mrekucci) [inline pkg/api/pin.go:58]: You can use "golang.org/x/sync/errgroup" instead and get rid of the mutex and error.

Comment (istae) [reply pkg/api/pin.go:58]: i thought about using that but I still need access to the internal error inside to goroutine to stop the traversal.

**Comment by istae** [comment]
it's capped at 100 parallel retrieval requests

**Comment by ldeffenb** [comment]
Ah, I see the weighted semaphore now!   That's what I get for commenting before actually studying, rather than skimming, the code!

**Pull request merged**